code/game/objects/items/toys.dm

```diff
@@ -536,7 +536,7 @@
 /obj/item/toy/snappop/Crossed(H as mob|obj)
 	if((ishuman(H))) //i guess carp and shit shouldn't set them off
 		var/mob/living/carbon/M = H
-		if(M.m_intent == "run")
+		if(M.m_intent != "walk")

```

Comment:
You could do a lot of good if you changed "run" and "walk" to DEFINES with integers.

```
#define WALK 0
#define RUN 1
#define SPRINT 2
```

As I can imagine BYOND not being the fastest with comparing strings.